hyppo/independence/hhg.py

```diff
@@ -129,16 +176,35 @@ def statistic(self, x, y):
         distx = x
         disty = y
 
-        if not self.is_distance:
-            distx, disty = compute_dist(
-                x, y, metric=self.compute_distance, **self.kwargs
-            )
-
-        S = _pearson_stat(distx, disty)
-        mask = np.ones(S.shape, dtype=bool)
-        np.fill_diagonal(mask, 0)
-        stat = np.sum(S[mask])
-        self.stat = stat
+        if not self.auto:
+            if not self.is_distance:
+                distx, disty = compute_dist(
+                    x, y, metric=self.compute_distance, **self.kwargs
+                )
+            S = _pearson_stat(distx, disty)
+            mask = np.ones(S.shape, dtype=bool)
+            np.fill_diagonal(mask, 0)
+            stat = np.sum(S[mask])
+            self.stat = stat
+
+        else:
+            # Fast HHG
+            if not self.is_distance:
```

Comment:
before you make into a helper function, rewrite this stat function like this:
```
if not self.is_distance:
     distx, disty = compute_dist(
         x, y, metric=self.compute_distance, **self.kwargs
     )

if self.auto:
     if not self.is_distance:
          pointer = (np.mean(x, axis=0), np.mean(y, axis=0))
          zx, zy = pointer
          zx = np.array(zx).reshape(1, -1)
          zy = np.array(zy).reshape(1, -1)
          xin = np.concatenate((zx, x))
          yin = np.concatenate((zy, y))
          distx = distx[0]
          distx = distx[distx != 0]
          disty = disty[0]
          disty = disty[disty != 0]
     stat = hoeffdings(distx, disty)
else:
     S = _pearson_stat(distx, disty)
     mask = np.ones(S.shape, dtype=bool)
     np.fill_diagonal(mask, 0)
     stat = np.sum(S[mask])
self.stat = stat
```

Reply:
Makes sense!

Reply:
Did the rewrite, along with the helper function replacement.